Client/Frontend/Login Management/BreachAlertsClient.swift

```diff
@@ -37,10 +60,21 @@ public class BreachAlertsClient: BreachAlertsClientProtocol {
                 completion(Maybe(failure: BreachAlertsError(description: error.localizedDescription)))
                 return
             }
-            guard let data = data, !data.isEmpty else {
-                completion(Maybe(failure: BreachAlertsError(description: "empty data")))
+            guard let data = data else {
+                completion(Maybe(failure: BreachAlertsError(description: "invalid data")))
                 return
             }
+
+            let httpResponse = response as? HTTPURLResponse
+            guard let etag = httpResponse?.allHeaderFields["Etag"] as Any as? String else { return }
+            guard let date = httpResponse?.allHeaderFields["Date"] as Any as? String else { return }
+
+            if profile.prefs.stringForKey(BreachAlertsClient.etagKey) != etag {
+                profile.prefs.setString(etag, forKey: BreachAlertsClient.etagKey)
```

Comment:
I am not sure why this isn't `Date.now()` tbh. Have to keep reading to see...

Reply:
I was thinking along the lines of, "the date is already here, I don't wanna call any functions I don't need to" but you're right in one of the below comments that if this one is used, then I need to format it for processing. It could go either way in that sense but I've figured out the `Date.now()` code if you'd like to switch to that